crds/virtualdisksnapshot.yaml

```diff
@@ -0,0 +1,119 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: virtualdisksnapshots.virtualization.deckhouse.io
+  labels:
+    heritage: deckhouse
+    module: virtualization
+spec:
+  group: virtualization.deckhouse.io
+  scope: Namespaced
+  names:
+    categories:
+      - all
+      - virtualization
+    plural: virtualdisksnapshots
+    singular: virtualdisksnapshot
+    kind: VirtualDiskSnapshot
+    shortNames:
+      - vdsnapshot
+      - vdsnapshots
+  preserveUnknownFields: false
+  versions:
+    - name: v1alpha2
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          description: |
+            Provides a resource for creating snapshots of existing virtual disks, which can be used as data sources for generating new virtual disks
+
+            A `VolumeSnapshot` is created under the hood of the resource.
+          required:
+            - spec
+          properties:
+            spec:
+              type: object
+              required:
+                - virtualDiskName
+                - volumeSnapshotClassName
+              properties:
+                virtualDiskName:
+                  type: string
+                  description: |
+                    The name of the virtual disk to take snapshot.
+                volumeSnapshotClassName:
+                  type: string
+                  description: |
+                    The name of the volume snapshot class to use while snapshotting virtual disk.
+                allowPotentiallyInconsistent:
+                  type: boolean
+                  default: false
+                  description: |
+                    Specify whether to allow the snapshotting of a virtual machine's disk that is attached to a running virtual machine without an agent, and as a result, cannot be frozen.
+
+                    If value is false, the snapshot of the virtual disk will be taken only in the following scenarios:
+                    - the virtual disk is not attached to any virtual machine.
+                    - the virtual disk is attached to a virtual machine that is powered off.
+                    - the virtual disk is attached to a virtual machine with an agent, and the freeze operation was successful.
+            status:
```

Comment:
There should be `consistent` param in status
with values: 
true - snapshot is consistent
nil - in other cases